pkg/build/strategies/layered/layered.go

```diff
@@ -141,8 +170,14 @@ func (b *Layered) Build(config *api.Config) (*api.Result, error) {
 	b.config.LayeredBuild = true
 	// new image name
 	b.config.BuilderImage = newBuilderImage
-	// the scripts are inside the image
-	b.config.ScriptsURL = "image://" + path.Join(getDestination(config), "scripts")
+	// see CreateDockerfile, conditional copy, location of scripts
+	scriptsIncluded := checkValidDirWithContents(path.Join(config.WorkingDir, api.UploadScripts))
+	glog.V(2).Infof("Scripts dir has contents %v", scriptsIncluded)
+	if scriptsIncluded {
+		b.config.ScriptsURL = "image://" + path.Join(getDestination(config), "scripts")
+	} else {
+		b.config.ScriptsURL = os.Getenv("STI_SCRIPTS_PATH")
```

Comment:
hm.  b.config.ScriptsURL should still be derived in the normal flow, meaning if the image has a label use that first, STI_SCRIPTS_PATH is considered deprecated.